### ISSUE TITLE
`no-invalid-regexp` does not allow "y"/"sticky" flag

### ISSUE THREAD

**Issue by ljharb** [eslint/eslint#1537]
Per https://people.mozilla.org/~jorendorff/es6-draft.html#sec-get-regexp.prototype.sticky "y" is a valid flag in ES6, and `new RegExp('a', 'y')` should not trigger `no-invalid-regexp` when the dev wants to allow that flag.

To avoid allowing "y" in pre-ES6 environments, perhaps this could be configurable in the rule, eg, a "flags" string/array?

(`eslint v0.10.0`)

**Comment by nzakas** [comment]
You can now enable this using the `ecmaFeatures` property in your config file:
https://github.com/eslint/eslint/tree/master/docs/configuring

(Checked into master)

**Comment by ljharb** [comment]
I'm running `eslint` off of the latest master, and I've added:
```
"ecmaFeatures": {
  "regexYFlag": true,
  "regexUFlag": true
}
```
to my `.eslintrc` file, but I'm still getting these errors. Is there something else I need to do to enable it?

**Comment by nzakas** [comment]
Hmmm, no, that should work. Reopening to investigate.

**Comment by ljharb** [comment]
If it becomes necessary I can push a temporary branch for you to look at, but I'd rather not since it's still in progress :-)

**Comment by nzakas** [comment]
Not necessary. Can you post what's output to the console and the line that has the regex on it?

**Comment by ljharb** [comment]
```bash
   95:15  error  Invalid flags supplied to RegExp constructor 'y'     no-invalid-regexp
   98:15  error  Invalid flags supplied to RegExp constructor 'u'     no-invalid-regexp
  119:15  error  Invalid flags supplied to RegExp constructor 'mygi'  no-invalid-regexp
  122:15  error  Invalid flags supplied to RegExp constructor 'mugi'  no-invalid-regexp
```

And the resulting lines:
```js
 95: expect(new RegExp('a', 'y').flags).to.equal('y');
 98: expect(new RegExp('a', 'u').flags).to.equal('u');
119: expect(new RegExp('a', 'mygi').flags).to.equal('gimy');
122: expect(new RegExp('a', 'mugi').flags).to.equal('gimu');
```

**Comment by nzakas** [comment]
Working on this.

**Comment by nzakas** [comment]
Depends on #1648, so switching to that.

**Comment by ljharb** [comment]
Hooray, you fixed it!

**Comment by nzakas** [comment]
Yup, and it was just pushed.

**Comment by ljharb** [comment]
To clarify: I saw the push, I immediately tested it, confirmed that it's fixed, and posted my comment :-) Thanks!
